atc/api/policychecker/handler.go

```diff
@@ -0,0 +1,49 @@
+package policychecker
+
+import (
+	"code.cloudfoundry.org/lager"
+	"context"
+	"net/http"
+
+	"github.com/concourse/concourse/atc/api/accessor"
+	"github.com/concourse/concourse/atc/policy"
+)
+
+func NewHandler(
+	logger lager.Logger,
+	handler http.Handler,
+	action string,
+	policyChecker policy.PreChecker,
+) http.Handler {
+	return policyCheckerHandler{
+		handler:       handler,
+		action:        action,
+		policyChecker: policyChecker,
+	}
+}
+
+type policyCheckerHandler struct {
+	logger        lager.Logger
+	handler       http.Handler
+	action        string
+	policyChecker policy.PreChecker
+}
+
+func (h policyCheckerHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	acc := accessor.GetAccessor(r)
+	ctx := context.WithValue(r.Context(), "accessor", acc)
+
+	pass, err := h.policyChecker.CheckHttpApi(h.action, acc.UserName(), r)
```

Comment:
Instead of getting the accessor and extracting the username here, could we move that logic into the implementation of `CheckHttpApi`? Just allows for fewer parameters/smaller interface.
```suggestion
	pass, err := h.policyChecker.CheckHttpApi(h.action, r)
```

Reply:
@pivotal-jamie-klassen I don't get this comment. To implement extracting username in `CheckHttpApi`, we have to pass `acc`, then it still have 3 parameters.